src/bootstrap/setup.rs

```diff
@@ -71,13 +82,39 @@ impl FromStr for Profile {
 
 impl fmt::Display for Profile {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            Profile::Compiler => write!(f, "compiler"),
-            Profile::Codegen => write!(f, "codegen"),
-            Profile::Library => write!(f, "library"),
-            Profile::User => write!(f, "user"),
-            Profile::Tools => write!(f, "tools"),
+        f.write_str(self.as_str())
+    }
+}
+
+impl Step for Profile {
+    type Output = ();
+    const DEFAULT: bool = true;
+
+    fn should_run(mut run: ShouldRun<'_>) -> ShouldRun<'_> {
+        for choice in Profile::all() {
+            run = run.alias(choice.as_str());
         }
+        run
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        let profile: Profile = run
+            .paths
+            .first()
+            .unwrap()
```

Comment:
This doesn't look right, it's valid to call `x.py setup` without arguments.

Hmm, but testing it locally it seems to work ok. Do you know why that is? And if so, can you record it as a comment?

Reply:
Please see if the added comments make sense, thanks!